subprojects/docs/src/docs/userguide/tasks.xml

```diff
@@ -290,52 +290,673 @@
+
+                    <tr>
+                        <td><literal>@OutputFiles</literal></td>
+                        <td><literal>FileCollection</literal>*</td>
+                        <td>A collection of output files (no directories)</td>
+                    </tr>
+
+                    <tr>
+                        <td><literal>@OutputDirectories</literal></td>
+                        <td><literal>FileCollection</literal>*</td>
+                        <td>A collection of output directories (no files)</td>
+                    </tr>
+
+                    <tr>
+                        <td><literal>@Nested</literal></td>
+                        <td>Any custom type</td>
+                        <td>A custom type that may not implement <literal>Serializable</literal> but does
+                            have at least one field or property marked with one of the annotations in this
+                            table. It could even be another <literal>@Nested</literal>.</td>
+                    </tr>
+
+                    <tr>
+                        <td><literal>@Console</literal></td>
+                        <td>Any type</td>
+                        <td>Indicates that the property is neither an input nor an output. It simply
+                            affects the console output of the task in some way.</td>
+                    </tr>
+
+                    <tr>
+                        <td><literal>@Internal</literal></td>
+                        <td>Any type</td>
+                        <td>Indicates that the property is used internally but is neither an input nor
+                            an output.</td>
+                    </tr>
+
+                    <tr>
+                        <td><literal>@SkipWhenEmpty</literal></td>
+
+                        <td></td>
+                        <td>Used with <literal>@InputFiles</literal> or <literal>@InputDirectory</literal>
+                            to tell Gradle to skip the task if the corresponding file collection or
+                            directory are empty.
+                        </td>
+                    </tr>
+                    <tfoot>
+                        <tr>
+                            <td colspan="3"><emphasis role="bold">*</emphasis> In fact, <literal>File</literal>
+                                can be any type accepted by <literal>Project.file()</literal> and
+                                <literal>FileCollection</literal> can be any type accepted by
+                                <literal>Project.files()</literal>. This includes instances of
+                                <literal>Callable</literal>, such as closures, allowing for lazy
+                                evaluation of the property values.
+                            </td>
+                        </tr>
+                    </tfoot>
+                </table>
+
+                <para>The last two annotations in the table are a special case as they don’t declare
```

Comment:
The last two are actually `@Internal` and `@SkipWhenEmpty`. I think you mean `@Console` and `@Internal` here. Might be best to just spell it out instead of backreferencing.